pkg/client/client.go

```diff
@@ -20,13 +21,13 @@ type BaseClient interface {
 	RefreshOptions(ctx context.Context, userOptions []string) error
 
 	// Status retrieves the workspace status
-	Status(ctx context.Context, options StatusOptions) (Status, error)
+	Status(ctx context.Context, options StatusOptions, cfg *config.Config) (Status, error)
```

Comment:
Looks like we need the config here in all of these functions to resolve the agent config, please correct me if I'm wrong.
If that's the case you can get a way with just resolving the value once over here: https://github.com/loft-sh/devpod/blob/c2855e63acf5bd9a8fe2fe545eee9f01392adfd0/pkg/options/resolve.go#L214
This way we wouldn't need to change the interface and all implementations for one specific line

Reply:
Thanks for pointing this out @pascalbreuninger! As it turns out the workspace client has a member variable containing the devPodConfig :disappointed: So I have simplified the "prop drilling" since lots of function now don't need this in the signature. The PR is up to date with this and is G2G in my eyes